#!/usr/bin/env python
PACKAGE = "pal_vision_segmentation"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("threshold", double_t, 0, "Pixels corresponding to farther distances (in m) will be discarded", 1.6, 0, 8.0)
gen.add("dilate_iterations", int_t, 0, "Dilate iterations", 7, 0, 20)
gen.add("dilate_size", int_t, 0, "Dilate kernel size", 5, 3, 15)
gen.add("erode_iterations", int_t, 0, "Erode iterations", 1, 0, 20)
gen.add("erode_size", int_t, 0, "Erode kernel size", 3, 3, 15)

exit(gen.generate(PACKAGE, "depth_segmentation_node", "DepthSegment"))
